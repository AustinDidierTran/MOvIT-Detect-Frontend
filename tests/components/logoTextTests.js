import React from 'react';
import PropTypes from 'prop-types';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import configureMockStore from 'redux-mock-store';
import toJson from 'enzyme-to-json';
import LogoText from '../../src/components/logoText';


Enzyme.configure({ adapter: new Adapter() });

describe('LogoText Tests', () => {
  const initialState = { applicationReducer: { language: 'en' } };
  const mockStore = configureMockStore();
  const store = mockStore(initialState);
  const props = {
    value: 'test',
    language: 'en',
  };

  it('should match the snapshot', () => {
    const wrapper = shallow(<LogoText store={store} {...props} />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should have proptypes', () => {
    // Actual value
    const actualValue = LogoText.propTypes;

    // Expected value
    const expectedValue = {
      placeHolder: PropTypes.string.isRequired,
      iconClass: PropTypes.string.isRequired,
      value: PropTypes.string,
      onChange: PropTypes.func.isRequired,
    };

    // Test
    expect(JSON.stringify(actualValue)).toEqual(JSON.stringify(expectedValue));
  });

  it('input prints the right value', () => {
    const value = 'test';
    const comp = shallow(
      <LogoText
        iconClass="fa fa-id-card"
        placeHolder="test"
        value={value}
        onChange={() => {}}
      />,
    );
    expect(comp.find('#logoText').props().value).toEqual(value);
  });

  it('input calls the method onChange with the right value', () => {
    let value = '';
    const change = (text) => { value = text; };
    const comp = shallow(
      <LogoText
        iconClass="fa fa-id-card"
        placeHolder="test"
        value={value}
        onChange={change}
      />,
    );
    comp.find('#logoText').simulate('change', { target: { value: 'test' } });
    expect(value).toEqual('test');
  });
});
