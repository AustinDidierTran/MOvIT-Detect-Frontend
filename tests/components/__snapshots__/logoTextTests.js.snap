// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogoText Tests should match the snapshot 1`] = `
<div
  className="form-group"
>
  <span
    className="col-md-1 col-md-offset-2 text-center"
    style={
      Object {
        "fontSize": "large",
        "paddingTop": "6px",
      }
    }
  >
    <i />
  </span>
  <div
    className="col-md-6"
  >
    <input
      className="form-control"
      id="logoText"
      onChange={[Function]}
      type="text"
      value="test"
    />
  </div>
</div>
`;
